packages/api/bin/serveUtils.js

```diff
@@ -82,12 +193,57 @@ async function addReconciliationReports(reconciliationReports) {
   );
 }
 
+async function addGranulesExecutions(granulesExecutions) {
```

Comment:
Can you help me understand what is calling this? I'm wondering if instead we can create the granuleExecution records as we create the granules or the execution records themselves

Reply:
Sure. On the dashboard integration tests we have a custom cypress plugin called `seedEverything`. Its purpose is to reset the dynamoDb and now Postgres tables between each suite of tests. We seed the database by emptying the tables and then reseeding each table in turn. We do seed granules and executions separately prior to calling `addGranulesExecutions`. We have a granulesFixture, executionsFixture, and I added granulesExecutionsFixture. I think you're asking if I could create the granules record, create the execution record, and then create the granules_executions record using the ID generated rather than having to look them up in a separate method. I believe I _could_. Truthfully, my goal was to have as little impact on existing integration tests as I could so I left the granulesFixture and executionsFixture exactly as they were and just added the granulesExecutionsFixture creating relationships  between the already existing granules and executions.

Reply:
So I think you can instead do:

- create executions first
- create granules, and as you're creating them, look up the execution PG record where `url` of the PG record matches the `granule.execution` value from the fixture. Then you can call `upsertGranuleWithExecutionJoinRecord` to write both the granule and granule execution records at the same time

https://github.com/nasa/cumulus/blob/master/packages/db/src/lib/granule.ts#L21

Then there should be no need to do a separate write or fixture for granulesExecutinos

Reply:
This was addressed as suggested.